### ISSUE TITLE
Gittyup requiring password for everything

### ISSUE THREAD

**Issue by Luxvao** [Murmele/Gittyup#795]
I'm using gittyup with github and the token. It's really inconvenient that gittyup always asks me for the password for everything. I don't want to keep my token available every time I use gittyup, especially because I can't see it again after viewing it once.

**Comment by Murmele** [comment]
Hi, which OS are you using?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/Murmele/Gittyup/issues/795#issue-2484745150

@Luxvao, have you changed credential storage from the default of "cache" to "store"?

**Comment by FabioLolix** [comment]
I had the same problem and following @RokeJulianLockhart tip solved it

**Comment by RokeJulianLockhart** [comment]
Thanks, @FabioLolix. @Luxvao, if that method remediates it for you too, please rename this issue to reflect that you want the *store* method to be the default, else this issue isn't actionable without you providing that @Murmele has requested.

On that note, by default, you should provide basic environment information in any issue you file.

**Comment by TaliesinToo** [comment]
FYI, I am using it on Windows, and to solve the same problem for me I had to set it to wincred instead. So perhaps we need different defaults for different OS installs?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/Murmele/Gittyup/issues/795#issuecomment-2610353352

@TaliesinToo:

1. Does `store` work *too*? I ask because I believe I use `store` on my Windows 11 Pro installation.

1. Irrespective, where did you get `wincred` from? I ask because I'd love to know whether there's something I can enumerate to get all the possible options.

   It might assist in diagnosing https://github.com/Murmele/Gittyup/issues/524#issue-1673912359.

**Comment by TaliesinToo** [comment]
> 1. Does `store` work _too_? I ask because I believe I use `store` on my Windows 11 Pro installation.

Nope, at least not with Gitlab. I tried both cache and store, with restarts of GittyUp in between, and both had the same constant prompting for username and password for every action that required talking to Gitlab. The only one that worked was wincred. I'm running Windows 10 Pro 22H2, and have Git For Windows 2.47.1 installed.
  
> 2. Irrespective, where did you get `wincred` from? I ask because I'd love to know whether there's something I can enumerate to get all the possible options.
>    It might assist in diagnosing [[Bug] Constant reprompting for HTTPS authentication credentials after installing `git-credential-manager`. #524 (comment)](https://github.com/Murmele/Gittyup/issues/524#issue-1673912359).

I just used the Tools -> Options menu, General tab, Credential store type dropdown, where those are the only three options listed.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/Murmele/Gittyup/issues/795#issuecomment-2610458886

@TaliesinToo, I just tried to clone a repository and hit https://github.com/Murmele/Gittyup/issues/827#issue-2807532192. However, I regardless appear to be authenticated with GitLab via an API key, since Gittyup can now enumerate my private repositories. Though, Gittyup was set to `manager` by default for me (albeit with the option of `wincred` too). Does `manager` appear for you too, and does it work for you too? For me too, switching to `wincred` didn't appear to revoke my authentication, and continued to work past a reinitialisation of Gittyup.

**Comment by TaliesinToo** [comment]
@RokeJulianLockhart I don't see the manager option, but I vaguely recall seeing that at one point. I think it might have been the default before I specified, and maybe it just lets the git credential manager choose the method. So I can't say for sure whether it worked, but if it had worked on manager, I would have had no reason to try the other options, so I'm guessing no.
